### ISSUE TITLE
[template/prefer-static-string-properties] ngSwitchCase should not trigger this rule

### ISSUE THREAD

**Issue by nicolashenry** [angular-eslint/angular-eslint#2237]
**Description and reproduction of the issue**

`template/prefer-static-string-properties` rule is reporting an error on ngSwitchCase, I think it should not happen.

[Reproduction](https://stackblitz.com/edit/angular-eslint-repros-vgmsptdb?file=src%2Fapp.component.html)

**Versions**

| package                                  | version |
| ---------------------------------------- | ------- |
| `@angular-eslint/eslint-plugin-template` | `19.1.0` |
| `@angular-eslint/template-parser`        | `19.1.0` |
| `@typescript-eslint/parser`              | `8.16.0` |
| `ESLint`                                 | `9.15.0` |
| `node`                                   | `22.13.1` |

```sh
Angular CLI: 19.0.0
Node: 18.20.3
Package Manager: npm 10.2.3
OS: linux x64

Angular: 19.0.0
... animations, build, cli, common, compiler, compiler-cli, core
... forms, platform-browser, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1900.0
@angular-devkit/core         19.0.0
@angular-devkit/schematics   19.0.0
@schematics/angular          19.0.0
rxjs                         7.8.1
typescript                   5.6.3
zone.js                      0.15.0
```
